### ISSUE TITLE
Incorrect scoping for chrome tracing format file

### ISSUE THREAD

**Issue by rgal** [jlfwong/speedscope#223]
When importing from a chrome tracing format file I see some problems with scoping.

![speedscope](https://user-images.githubusercontent.com/8567076/60436645-616f5800-9c04-11e9-8214-97205b801070.PNG)

The difference between good.json and bad.json is just the order of events in the file. good.json has the events in ascending timestamp order, bad.json has two events out of order. The trace event format specification says that "The events do not have to be in timestamp-sorted order." so this shouldn't matter. Both display fine in chrome://tracing.

[good.json.txt](https://github.com/jlfwong/speedscope/files/3345555/good.json.txt)
[bad.json.txt](https://github.com/jlfwong/speedscope/files/3345558/bad.json.txt)

The original data is from LLVM -ftime-trace with some additional processing afterwards.

**Comment by jlfwong** [comment]
Thanks for providing the test cases! If the description is accurate, then this indeed seems like a bug

**Comment by rgal** [comment]
Thanks. A few more notes. "B" and "C" both finish at exactly the same time. There are some "Event discarded" warnings on the Console from bad.json. I can't think of a reason why events finishing at the same time shouldn't be allowed.

**Comment by rgal** [comment]
It also seems to matter what the "name" of the event is. Two events of the same "name" finishing at the same time works okay, but with different names this causes problems.

**Comment by rgal** [comment]
I think that this is a regression since v1.5.0 so was caused by https://github.com/jlfwong/speedscope/commit/0c1e477f359e1702f73952f46ff708eda920e300.

I don't see the bug with that version and can get it to work on v1.5.1 by removing the check "&& lastEvent.name == ev.name" in trace-events.ts.

I attached some simpler examples:
[good.json.txt](https://github.com/jlfwong/speedscope/files/3520821/good.json.txt)
[bad.json.txt](https://github.com/jlfwong/speedscope/files/3520822/bad.json.txt)

**Comment by thomasjm** [comment]
I'm seeing something very similar to this, but it's not from importing another format. 

I'm trying to generate data in speedscope's native format and as far as I know the files I'm generating are well-formed. I attached an example of this.

For example, look at frame `0`, with name `"Requirements tests/should have a cpp11 kernel"`. If you look at the events in the profile, it is opened in the first event and then closed a mere 5 seconds later. But if you look at what speedscope renders, that frame goes much longer, across the entire top of the profile:

![Screenshot from 2020-05-26 06-21-07](https://user-images.githubusercontent.com/1634990/82905844-37f4a480-9f19-11ea-8760-6afab2c13aad.png)

In more complicated examples, I can observe this kind of over-large stack frame happening many times. Would greatly appreciate any suggestions for tracking this down.

[speedscope.json.txt](https://github.com/jlfwong/speedscope/files/4682453/speedscope.json.txt)

**Comment by jlfwong** [comment]
@thomasjm That file isn't well-formed, but for reasons that speedscope wasn't properly complaining about until #273 (which hasn't yet been deployed to https://www.speedscope.app/).

When I drop that file onto speedscope in latest master it says the following:

```
Failed to load format Error: Tried to leave frame "error video" while frame "individual spec" was at the top at 18.45978808403015
    at CallTreeProfileBuilder._leaveFrame (profile.ts:608)
    at CallTreeProfileBuilder.leaveFrame (profile.ts:632)
    at importEventedProfile (file-format.ts:117)
    at importSpeedscopeProfile (file-format.ts:152)
    at file-format.ts:162
    at Array.map (<anonymous>)
    at importSpeedscopeProfiles (file-format.ts:162)
    at profile.proto.js:3543
    at Generator.next (<anonymous>)
    at fulfilled (profile.proto.js:3543)
```

See #272 for an explanation of why this is ill-formed.

The issue reported by @rgal is different. I still sadly haven't had the time to dig into fixing that issue.

**Comment by thomasjm** [comment]
Aha, never mind then. Thanks for the quick response @jlfwong !

**Comment by jlfwong** [comment]
@rgal This is, uh, pretty late, but I fixed this along with a few other trace event related bugs, and the files from this issue helped me debug the problem. The fix is published to npm as speedscope@1.11.1 and it's live on https://www.speedscope.app/
